packages/fx-core/src/component/generator/templates/templateGenerator.ts

```diff
@@ -0,0 +1,223 @@
+    destinationPath: string,
+    actionContext?: ActionContext
+  ): Promise<Result<undefined, FxError>> {
+    return Promise.resolve(ok(undefined));
+  }
+
+  public getDefaultReplaceMap(inputs: Inputs): { [key: string]: string } {
+    const appName = inputs[QuestionNames.AppName] as string;
+    const safeProjectName =
+      inputs[QuestionNames.SafeProjectName] ?? convertToAlphanumericOnly(appName);
+    const targetFramework = inputs.targetFramework;
+    const placeProjectFileInSolutionDir = inputs.placeProjectFileInSolutionDir === "true";
+    const llmService: string | undefined = inputs[QuestionNames.LLMService];
+    const openAIKey: string | undefined = inputs[QuestionNames.OpenAIKey];
+    const azureOpenAIKey: string | undefined = inputs[QuestionNames.AzureOpenAIKey];
+    const azureOpenAIEndpoint: string | undefined = inputs[QuestionNames.AzureOpenAIEndpoint];
+
+    return {
+      appName: appName,
+      ProjectName: appName,
+      TargetFramework: targetFramework ?? "net8.0",
+      PlaceProjectFileInSolutionDir: placeProjectFileInSolutionDir ? "true" : "",
+      SafeProjectName: safeProjectName,
+      SafeProjectNameLowerCase: safeProjectName.toLocaleLowerCase(),
+      enableTestToolByDefault: enableTestToolByDefault() ? "true" : "",
+      useOpenAI: llmService === "llm-service-openai" ? "true" : "",
+      useAzureOpenAI: llmService === "llm-service-azure-openai" ? "true" : "",
+      openAIKey: openAIKey ?? "",
+      azureOpenAIKey: azureOpenAIKey ?? "",
+      azureOpenAIEndpoint: azureOpenAIEndpoint ?? "",
+      isNewProjectTypeEnabled: isNewProjectTypeEnabled() ? "true" : "",
+      NewProjectTypeName: process.env.TEAMSFX_NEW_PROJECT_TYPE_NAME ?? "TeamsApp",
+      NewProjectTypeExt: process.env.TEAMSFX_NEW_PROJECT_TYPE_EXTENSION ?? "ttkproj",
+    };
+  }
+
+  private getTemplateName(inputs: Inputs) {
+    const language = inputs[QuestionNames.ProgrammingLanguage];
+    const capability = inputs.capabilities as string;
+    const meArchitecture = inputs[QuestionNames.MeArchitectureType] as string;
+    const apiMEAuthType = inputs[QuestionNames.ApiMEAuth] as string;
+    const trigger = inputs[QuestionNames.BotTrigger] as string;
+    let feature = `${capability}:${trigger}`;
+    if (
+      capability === CapabilityOptions.m365SsoLaunchPage().id ||
+      capability === CapabilityOptions.m365SearchMe().id
+    ) {
+      inputs.isM365 = true;
+    }
+
+    if (
+      language === "csharp" &&
+      capability === CapabilityOptions.notificationBot().id &&
+      inputs.isIsolated === true
+    ) {
+      feature += "-isolated";
+    }
+
+    if (meArchitecture) {
+      feature = `${feature}:${meArchitecture}`;
+    }
+    if (
+      inputs.targetFramework &&
+      inputs.targetFramework !== "net6.0" &&
+      inputs.targetFramework !== "net7.0" &&
+      (capability === CapabilityOptions.nonSsoTab().id || capability === CapabilityOptions.tab().id)
+    ) {
+      feature = `${capability}:ssr`;
```

Comment:
Looks most feature map use capability and custom options, is it easy to be refactor as real map table?

Reply:
Yes. I will refine the map logic in next PR. This PR focus on the implementation of new template generator class.